robot/web/web.go

```diff
@@ -452,21 +454,21 @@ func (svc *webService) updateResources(resources map[resource.Name]interface{})
 func (svc *webService) Stop() {
 	svc.mu.Lock()
 	defer svc.mu.Unlock()
-	if svc.cancelFunc != nil {
-		svc.cancelFunc()
-		svc.cancelFunc = nil
+	for _, cancel := range svc.cancelFuncs {
+		cancel()
 	}
+	svc.cancelFuncs = nil
 }
 
 // Close closes a webService via calls to its Cancel func.
 func (svc *webService) Close() error {
 	svc.mu.Lock()
 	defer svc.mu.Unlock()
 	var err error
-	if svc.cancelFunc != nil {
-		svc.cancelFunc()
-		svc.cancelFunc = nil
+	for _, cancel := range svc.cancelFuncs {
+		cancel()
 	}
+	svc.cancelFuncs = nil
```

Comment:
Shouldn't svc.isRunning be set to false somewhere around here? (Probably isn't used after this point, but for consistency.)

Reply:
Yep! Done.